zeusd/src/devices/cpu/mod.rs

```diff
@@ -0,0 +1,170 @@
+            // Channel to send commands to the CPU management task.
+            let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
+            senders.push(tx);
+            // The CPU management task will automatically terminate
+            // when the server terminates and the last sender is dropped.
+            tokio::spawn(cpu_management_task(cpu, rx));
+            tracing::info!("Background task for CPU {} successfully spawned", cpu_id);
+        }
+        Ok(Self { senders })
+    }
+
+    pub async fn send_command_blocking(
+        &self,
+        cpu_id: usize,
+        command: CpuCommand,
+        request_start_time: Instant,
+    ) -> Result<RaplResponse, ZeusdError> {
+        if cpu_id >= self.senders.len() {
+            return Err(ZeusdError::CpuNotFoundError(cpu_id));
+        }
+        let (tx, mut rx) = tokio::sync::mpsc::channel(1);
+        self.senders[cpu_id]
+            .send((command, Some(tx), request_start_time, Span::current()))
+            .unwrap();
+        match rx.recv().await {
+            Some(result) => result,
+            None => Err(ZeusdError::CpuManagementTaskTerminatedError(cpu_id)),
+        }
+    }
+
+    pub async fn stop_monitoring(&self) {
+        for sender in self.senders.clone() {
+            let (tx, _) = tokio::sync::mpsc::channel(1);
+            sender
+                .send((
+                    CpuCommand::StopMonitoring {},
+                    Some(tx),
+                    Instant::now(),
+                    Span::current(),
+                ))
+                .unwrap();
+        }
+    }
+}
+
+/// A CPU command that can be executed on a CPU.
+#[derive(Debug)]
+pub enum CpuCommand {
+    /// Get the CPU and DRAM energy measurement for the CPU index
+    GetIndexEnergy {
+        cpu: bool,
+        dram: bool,
+    },
+    StopMonitoring {},
+}
+
+async fn cpu_management_task<T: CpuManager>(
+    mut cpu: T,
+    mut rx: UnboundedReceiver<CpuCommandRequest>,
+) {
+    while let Some((command, response, start_time, span)) = rx.recv().await {
+        let _span_guard = span.enter();
+        let result = command.execute(&mut cpu, start_time);
+        if let Some(response) = response {
+            if response.send(result).await.is_err() {
+                tracing::error!("Failed to send response to caller");
+            }
+        }
+    }
+}
+
+impl CpuCommand {
+    fn execute<T>(
+        &self,
+        device: &mut T,
+        _request_arrival_time: Instant,
+    ) -> Result<RaplResponse, ZeusdError>
+    where
+        T: CpuManager,
+    {
+        match *self {
+            Self::GetIndexEnergy { cpu, dram } => {
+                let mut cpu_energy_uj = None;
+                let mut dram_energy_uj = None;
+                if cpu {
+                    cpu_energy_uj = Some(device.get_cpu_energy()?);
+                }
+                if dram {
+                    dram_energy_uj = device.get_dram_energy()?;
+                }
```

Comment:
This seems weird. I guess `get_dram_energy` returns an `Option` and `get_cpu_energy` returns a number. I presume that is because DRAM energy measurement may not be supported on the particular platform, but now you ended up normalizing everything to `Option` here anyway.

Instead, it would be cleaner for both functions to just return a Result (raw energy number if successful and `ZeusdError` if not), and higher-level code (e.g., router handler or slightly below) ensures that `dram` is never `true` when the underlying platform does not support DRAM energy measurement.

Side note, something like this would be more conventional for Rust:
```rust
let cpu_energy_uj = if cpu { device.get_cpu_energy()? } else { None };
```